# <font face="garamond" size="18" color="#122DAC">***Replication Report on "A five-factor asset pricing model" (Fama & French,  2015)***</font>
## <font face="garamond" size="6" color="#122DAC">*Econ 430*</font> 
### <font face="garamond" size="6" color="#122DAC">*Jacob Williams, Josh Kentworthy, and Ignacio Ramirez*</font>

## <font face="garamond" size="6" color="#black">*University of California Los Angeles*</font> #
## <font face="garamond" size="6" color="#black">*November, 20 2025*</font>

### **Original Paper:** Fama, E. F., & French, K. R. (2015). A five-factor asset pricing model. *Journal of Financial Economics, 116*(1), 1-22.

# 1. Original Study Summary

## (a) Research Question: Clearly state the paper’s main research question and economic Motivation

Modern asset-pricing began with a simple premise: if markets are competitive and information flows freely, prices should adjust to news and leave no predictable profit opportunities. This idea is what became the **Efficient Market Hypothesis (EMH)**. EMH emerged in the 1950s and 60s out of early empirical work showing that stock returns had almost no autocorrelation, suggesting prices behave like a fair game responding only to new information.

A decade later, Sharpe (1964) and Lintner (1965) formalized the first equilibrium model linking risk to expected returns: the **Capital Asset Pricing Model (CAPM)**. Its prediction was:

**(1) CAPM Formula**  
\[$
E[R_i] - R_f = \beta_i\,(E[R_M] - R_f)
$\]

With **beta** as the sole predictor of cross-sectional differences in average returns. For 20 years this model remained empirically reasonable.

But as Fama repeatedly emphasizes, CAPM’s “golden age” was short. By the early 1980s and into the 1990s, empirical papers accumulated showing that beta alone could not explain average returns sorted on firm **size** (Banz 1981), **earnings-price ratios** (Basu 1977), **leverage** (Bhandari 1988), or **book-to-market** (Rosenberg, Reif & Lanstein 1985). Additional patterns such as long-term reversals (DeBondt & Thaler 1985) and the January effect (Keim 1983) reinforced the conclusion that CAPM missed persistent dimensions of return variance.  

These anomalies eventually shifted professional consensus. As the *New York Times* summarized in 1992—reflecting the conclusions of Fama and French—“beta as the sole variable explaining returns on stocks is dead.” (Berg 1992)

These empirical failures exposed a deeper theoretical problem: CAPM offered too narrow a representation of risk. Asset-pricing models required additional dimensions to explain why some portfolios earned persistently higher returns than others.

When Eugene Fama and Kenneth French aimed at capturing these missing dimensions, they developed the **Fama–French Three-Factor Model (1993)**. In it they augmented the market factor with size (SMB: small minus big) and value (HML: high minus low book-to-market):

**(2) Fama–French Three-Factor Formula**  
\[$
R_i - R_f = \alpha_i + \beta_i (R_M - R_f) + s_i\,\text{SMB} + h_i\,\text{HML} + \varepsilon_i
$\]

What the Three-Factor Model aims to answer is: can size (SMB) and value (HML) jointly explain the cross-section of average returns better than CAPM?

---

After the Three-Factor Model (1993), Fama and French showed that SMB and HML captured important patterns that CAPM could not explain. But the model was still incomplete: it summarized the data, yet it did not explain **why** these patterns existed. Subsequent work showed that **profitability** and **investment** were also strongly related to average returns and were not absorbed by the three factors.

The motivation for revisiting the model comes directly from the **dividend discount model**, which expresses a stock’s price as the present value of expected future dividends:

**(3) Dividend Discount Model**  
\[$
P_t = \sum_{j=1}^{\infty} \frac{E_t[D_{t+j}]}{(1+r)^j}
$\]

The DDM can be rewritten in terms of future earnings and changes in book equity:

**(4) \(M_t\)**  
\[$
M_t = \frac{B_{t+1} - B_t + D_{t+1}}{E_t[R_{t+1}]}
$\]

This implies three core relationships:
- A higher book-to-market ratio is associated with **higher expected returns**  
- Higher expected profitability is associated with **higher expected returns**  
- Higher expected investment (growth in book equity) is associated with **lower expected returns**

This leads directly to the research question in the **Five-Factor Model of Fama and French (2014):**

> **“Do profitability and investment add explanatory power beyond the three-factor model?”**

The economic interpretation is straightforward: if expected returns are tied to how firms earn profits and reinvest capital, then an empirical model that omits these drivers is incomplete. The purpose of the five-factor model is to test whether incorporating **profitability (RMW)** and **investment (CMA)** produces a model better aligned with underlying expected returns.


## (b) Theoretical Framework: Summarize the economic theory underlying the analysis


The primary economic theory that the paper's analysis is based on is the dividend discount model. The authors state this directly: "We can use the dividend discount model to explain why these variables are related to average returns.” (Fama and French, 2014, p. 1)

This model states that the price of the stock is the discounted value of its expected future dividends. The authors then mathematically rearrange this model into a valuation equation (Equation 3) that expresses the ratio of a firm's market value to its book value (Mt /Bt ) as a function of its expected future earnings and expected future investments (changes in book equity) .

This valuation equation provides the theoretical justification for the new factors by generating three specific predictions about expected returns . As the paper states:

“Eq. (3) makes three statements about expected stock returns. First, fix everything in (3) except the current value of the stock, Mt , and the expected stock return, r. Then a lower value of Mt , or equivalently a higher book-to-market equity ratio, Bt /Mt , implies a higher expected return. Next, fix Mt  and the values of everything in (3) except expected future earnings and the expected stock return. The equation then tells us that higher expected earnings imply a higher expected return. Finally, for fixed values of Bt , Mt , and expected earnings, higher expected growth in book equity investment implies a lower expected return." (Fama and French, 2014, p. 2)

In short, the dividend discount model provides the theoretical basis for why value (high B/M), profitability (high earnings), and investment (low investment) should all be related to higher average stock returns. These three predictions are the "natural choices" that motivate adding profitability and investment factors to the model.	

## (c) Methodology: Describe the econometric approach, model specification, and estimation technique

The paper's core model specification is the five-factor asset pricing model. The authors explicitly define this in a regression equation:
"Motivated by this evidence and the valuation Eq. (3), we add profitability and investment factors to the three-factor model:
$R_{it}-R_{Ft}=a_{i}+b_{i}(R_{Mt}-R_{Ft})+s_{i}SMB_{t}+h_{i}HML_{t}+r_{i}RMW_{t}+c_{i}CMA_{t}+e_{it}$" (Fama and French, 2014, p. 3)

In this equation, the dependent variable ($R_{it}−R_{Ft}$) is the excess return on a portfolio or security i at time t. The independent variables are the five factors: the market excess return ($R_{Mt}−R_{Ft}$), the size factor ($SMBt$), the value factor ($HMLt$), the profitability factor ($RMWt$), and the investment factor ($CMAt$).
The estimation technique used is a time-series regression. This regression is run for each of the left-hand-side ($LHS$) portfolios (e.g., the 25 Size-B/M portfolios) against the five factor returns on the right-hand-side ($RHS$). The output of this estimation provides the factor loadings (the b, s, h, r, and c coefficients) for each portfolio, as well as the intercept, $a_i$.

The econometric approach is designed to test how well the model explains average returns. The central hypothesis of this approach is that the model's factors should capture all systematic variation in average returns, leaving no unexplained portion. The paper states this testable implication clearly:
"If the exposures to the five factors, $b_i$, $s_i$, $h_i$, $r_i$, and $c_i$, capture all variation in expected returns, the intercept $a$ in (5) is zero for all securities and portfolios i." (Fama and French, 2014, p. 3)

Therefore, the primary goal is to test whether these intercepts ($a_i$), also known as "alphas," are statistically indistinguishable from zero. To do this for a set of portfolios simultaneously, the authors employ a specific, formal statistical test, as noted here:
"Table 5 shows the GRS statistic of Gibbons, Ross, and Shanken (1989) that tests this hypothesis for combinations of LHS portfolios and factors." (Fama and French, 2014, p. 9)

The estimation technique used to obtain the intercepts and factor exposures is identified as a time-series regression (this quote refers to the three-factor model, but the same technique is applied to the five-factor model):
“Tests of the three-factor model center on the time-series regression:
$R_{it}-R_{ft}=a_i+b_i(R_{mt}-R_{ft})+s_iSMB_t+h_iHML_t+e_{it}$” (Fama and French, 2014, p. 3)


## (d) Key Variables: Define dependent and independent variables with their economic interpretation

The dependent variable in the regressions is the **excess return on a security or portfolio ($R_{it}−R_{Ft}$).**
In this equation $R_{it}$ is the return on security or portfolio $i$ for period $t$, $R_{FΩ}$ is the riskfree returnn.

The independent variables are the five factors used to explain the dependent variable's returns.


**Market Factor ($R_{Mt}−R_{Ft}$):**
"$R_{Mt}$ is the return on the value-weight (VW) market portfolio..." (subtracted by the risk-free return, $R_{Ft}$).

- **Economic Interpretation:** This captures the overall market risk.


**Size Factor ($SMB$):**
"$SMB_t$ is the return on a diversified portfolio of small stocks minus the return on a diversified portfolio of big stocks..."

- **Economic Interpretation:** This captures the "size premium," or the difference in returns between small-cap and large-cap firms.


**Value Factor ($HML$):**
"$HML_t$ is the difference between the returns on diversified portfolios of high and low B/M stocks..."

- **Economic Interpretation:** This captures the "value premium," or the difference in returns between firms with high book-to-market ratios (value stocks) and low book-to-market ratios (growth stocks).


**Profitability Factor ($RMW$):**
"$RMW_t$ is the difference between the returns on diversified portfolios of stocks with robust and weak profitability..."

- **Economic Interpretation:** This captures the "profitability premium," or the difference in returns between firms with high (robust) profitability and low (weak) profitability.


**Investment Factor ($CMA$):**
"$CMA_t$ is the difference between the returns on diversified portfolios of the stocks of low and high investment firms, which we call conservative and aggressive."

- **Economic Interpretation:** This captures the "investment premium," or the difference in returns between firms that invest conservatively (low investment) and those that invest aggressively (high investment).


## (e) Main Findings: Summarize the authors’ primary empirical results and conclusions

The authors' primary empirical finding is that their new model is a clear improvement over their previous one. As they state, "A five-factor model directed at capturing the size, value, profitability, and investment patterns in average stock returns performs better than the three-factor model of Fama and French (FF, 1993)".


However, this improved performance does not mean the model is perfect. While it is a better description of returns, it is still statistically rejected as a complete model. The authors write, "The GRS test easily rejects a five-factor model directed at capturing these patterns, but we estimate that the model explains between 71% and 94% of the cross-section variance of expected returns for the Size, B/M, OP, and Inv portfolios we examine". (Fama and French, 2014, p. 9)


One of the most striking conclusions is that the addition of the new profitability and investment factors makes the original value factor (HML) unnecessary for explaining average returns.
- "With the addition of profitability and investment factors, the value factor of the FF three-factor model becomes redundant for describing average returns in the sample we examine." (Fama and French, 2014, p. 1)


The reason for this redundancy is that the information captured by the value factor is also captured by the other factors in the model.
- "The average HML return is captured by the exposures of HML to other factors". (Fama and French, 2014, p. 2) The authors elaborate on this, concluding, "...our results suggest that HML is a redundant factor in the sense that its high average return is fully captured by its exposures to $R_M−R_F$, SMB, and especially RMW and CMA." (Fama and French, 2014, p. 19)


Despite its successes, the authors clearly identify the model's main failure point, which they term its "lethal" problem.
- "The five-factor model's main problem is its failure to capture the low average returns on small stocks whose returns behave like those of firms that invest a lot despite low profitability."(Fama and French, 2014, p. 1)


This specific type of stock portfolio consistently produces results that the model cannot explain.
- "One of our more interesting results is that portfolios of small stocks with negative exposures to RMW and CMA are the biggest asset pricing problem in four of the six sets of LHS portfolios examined here."(Fama and French, 2014, p. 19)


From this, the authors draw a firm conclusion about the nature of this "lethal" portfolio: 
- "...we conclude that the lethal portfolios contain small stocks of firms that invest a lot despite low profitability". (Fama and French, 2014, p. 19)


## (f) Contribution: Explain what gap in the literature this paper addresses

___

# 2. Data Acquisition and Description

## (a) Data Sources

All data for this replication project comes from the **Kenneth R. French Data Library**, the public data archive maintained by the co-author of the Five-Factor Asset Pricing Model. The library provides the exact factor and portfolio files used in Fama and French (2015), already cleaned and formatted by the authors. Because the data is published directly by the original source, no external cleaning or reconstruction was required.

### Five-Factor Model Data
The five factor series—Mkt-RF, SMB, HML, RMW, CMA, and RF—was downloaded from the:

**Kenneth R. French Data Library**  
https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html

We use the **Changes in CRSP Data** monthly dataset, which contains the market excess return and the four additional factors introduced in the paper. The file covers data from July, 1963 onward and is updated regularly by the authors.

### Test Portfolio Data
To replicate the empirical setup in the original study, we also use the **`25 Portfolios Formed on Size and Book-to-Market (5x5)`** dataset from the same website. These portfolios are constructed using two-way sorts on firm size and book-to-market ratios and are the standard test assets used in evaluating factor models.

### Structure of the Author-Provided Files
Both datasets follow a consistent CSV format:

- Several descriptive header lines  
- A date column labeled "Unnamed: 0" containing YYYYMM date identifiers  
- Monthly percentage returns for each factor or portfolio  
- A copyright footer line  

Since these files are prepared by the authors themselves, they require almost no cleaning beyond removing the non-data header and footer lines. After these are removed, the files load correctly and can be merged using the date column.

## (a) Data Sources

All data for this replication project comes from the **Kenneth R. French Data Library**, the public data archive maintained by the co-author of the Five-Factor Asset Pricing Model. The library provides the exact factor and portfolio files used in Fama and French (2015), already cleaned and formatted by the authors. Because the data is published directly by the original source, no external cleaning or reconstruction was required.

### Five-Factor Model Data
The five factor series—Mkt-RF, SMB, HML, RMW, CMA, and RF—was downloaded from the:

**Kenneth R. French Data Library**  
https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html

We use the **Fama/French 5 Factors (2x3)** monthly dataset, which contains the market excess return and the four additional factors introduced in the paper. The file covers data from July 1963 onward and is updated regularly by the authors.

### Test Portfolio Data
To replicate the empirical setup in the original study, we also use the **`25 Portfolios Formed on Size and Book-to-Market (5x5)`** dataset from the same website. These portfolios are constructed using two-way sorts on firm size and book-to-market ratios and are the standard test assets used in evaluating factor models.

### How the Authors Source Their Underlying Data
While our replication uses the already-constructed CSV files published on French’s website, the **raw data** used by Fama and French in the original study comes from two institutional databases.  
Stock returns—including prices, total shares, delisting returns, and market equity—are obtained from the **CRSP (Center for Research in Security Prices)** database. Firm-level accounting variables such as book equity, operating profitability, and total assets are sourced from **Compustat**. The authors merge CRSP and Compustat data each June to form portfolios sorted by size, value, profitability, and investment. These portfolios are then used to compute the SMB, HML, RMW, and CMA factor returns. The factor and portfolio CSV files available on the Data Library are constructed using these procedures, which is why we do not need to rebuild the factors from raw CRSP or Compustat data ourselves.

### Structure of the Author-Provided Files
Both datasets follow a consistent CSV format:

- Several descriptive header lines  
- A date column labeled `"Unnamed: 0"` containing YYYYMM date identifiers  
- Monthly percentage returns for each factor or portfolio  
- A copyright footer line  

Since these files are prepared by the authors themselves, they require almost no cleaning beyond removing the non-data header and footer lines. After these are removed, the files load correctly and can be merged using the date column.


## (b) Data Processing

Because the datasets are published directly by Kenneth French, they require minimal cleaning. The main steps involved removing descriptive header and footer rows, renaming the date column, setting the date as the index, and merging the factor data with the portfolio returns so the regression could be estimated.

### Preparing the Five-Factor File
The five-factor CSV contains several descriptive header lines followed by a numerical dataset, along with a copyright footer at the bottom. These non-data rows were removed on import. The date column appears as "Unnamed: 0" in the raw file, so it was renamed to "date" and used as the index.

```python
fdata = pd.read_csv('fdata.csv', skiprows=4)
fdata = fdata[pd.to_numeric(fdata['Unnamed: 0'], errors='coerce').notna()]
fdata = fdata.rename(columns={'Unnamed: 0': 'date', 'Mkt-RF': 'MKTRF'})
fdata = fdata.set_index('date')
```
### Preparing the 25 Portfolios (5x5) File

The file containing the **25 Size–Book-to-Market portfolios (5x5)** is stored locally as **`data55.csv`**. This dataset includes 25 portfolio return series created from two-way sorts on firm size and book-to-market ratios. Similar to the factor file, it includes a long descriptive header and a non-numerical footer line.

These rows were removed by skipping the header on import and slicing off the final line. The date column was again labeled "Unnamed: 0" in the raw file and was renamed for consistency before setting it as the index.

```python
data55 = pd.read_csv('data55.csv', skiprows=15)
data55 = data55.rename(columns={'Unnamed: 0': 'date'})
data55 = data55.set_index('date')
data55 = data55[:-1]
```

This results in a clean monthly return series for all 25 portfolios, indexed by the same YYYYMM date format used in the factor dataset.

### Merging the Data

Once both datasets were cleaned and indexed by date, they were merged using an inner join. This ensures that each portfolio return observation is matched with the corresponding month’s factor values from the five-factor model.

```python
data = pd.merge(fdata, data55, on='date', how='inner')
```

The merged dataset contains the full set of portfolio returns along with the market, size, value, profitability, and investment factors. This final table is the basis for the regression analysis presented in Section 3.

### (c) Sample Description:

In [15]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from statsmodels.regression.linear_model import OLS
import requests # This is the library for making HTTP requests. In this notebook, the module is grabbing data from Fama-French data library URL and downloading the .zip file. 
import zipfile # This is the built-in Python module for reading and writing ZIP archives. In this notebook,  after the requests library downloads the file, zipfile is used to open the .zip archive while it's still in memory. This allows us to extract the zip files.
from io import BytesIO, TextIOWrapper # This is a "translator" or "decoder." It takes a stream of raw bytes and turns it into a stream of readable text (strings). In this notebook, .txt filez inside the archive that is still just bytes, TextIOWrapper is used to "wrap" that byte stream, allowing us to read it as regular text.

In [11]:
fdata = pd.read_csv('factor_data.csv', skiprows=4)
fdata = fdata[pd.to_numeric(fdata['Unnamed: 0'], errors='coerce').notna()]
fdata = fdata.rename(columns = {'Unnamed: 0': 'date', 'Mkt-RF':'MKTRF'})
fdata = fdata.set_index('date')

# Converting data type to numeric
variables_to_convert = ['MKTRF', 'SMB', 'HML', 'RMW', 'CMA', 'RF']
fdata[variables_to_convert] = fdata[columns_to_convert].apply(pd.to_numeric, errors='coerce')

fdata

MKTRF    SMB    HML    RMW    CMA    RF
date                                           
196307  -0.39  -0.48  -0.81   0.64  -1.15  0.27
196308   5.08  -0.80   1.70   0.40  -0.38  0.25
196309  -1.57  -0.43   0.00  -0.78   0.15  0.27
196310   2.54  -1.34  -0.04   2.79  -2.25  0.29
196311  -0.86  -0.85   1.73  -0.43   2.27  0.27
...       ...    ...    ...    ...    ...   ...
  2020  23.65   3.74 -46.94  -5.21 -11.70  0.44
  2021  23.87  -1.02  25.61  26.72  11.67  0.04
  2022 -21.27  -1.57  25.68   6.55  22.50  1.42
  2023  21.75  -6.31 -13.98   6.24 -20.61  4.95
  2024  19.79 -13.52  -8.71   5.03  -9.83  5.26

[808 rows x 6 columns]

In [13]:
fdata.describe()

MKTRF         SMB         HML         RMW         CMA          RF
count  808.000000  808.000000  808.000000  808.000000  808.000000  808.000000
mean     1.123639    0.367475    0.532290    0.517401    0.459988    0.676584
std      6.713008    4.716616    5.219738    3.463540    3.541550    1.430856
min    -38.250000  -27.750000  -46.940000  -28.940000  -20.610000    0.000000
25%     -2.015000   -1.677500   -1.492500   -0.862500   -1.060000    0.160000
50%      1.090000    0.045000    0.230000    0.285000    0.120000    0.390000
75%      3.870000    2.212500    2.060000    1.510000    1.580000    0.540000
max     35.180000   50.180000   45.280000   26.720000   30.010000   14.710000

This analysis uses monthly returns for the Fama-French five factors: market excess return ($R_M−R_F$), size ($SMB$), value ($HML$), profitability ($RMW$), and investment ($CMA$), plus the risk-free rate ($R_F$). The descriptive statistics provided in the paper can be found at page 7 in table 4, panel A (2x3 Factors).

Our samples contain 808 monthly observations, which is significantly larger than the 606 monthly observations used in the original Fama and French (2015) paper. The paper's samples cover the period from July 1963 to December 2013. The 202-month difference is simply due to the use of a more recent dataset.

**Descriptive Statistics Comparison**

The table below compares the summary statistics from my sample (N=808) to those reported in the original paper (N=606). All values are presented as monthly percentages.

| Variable | Statistic | Our Samples (N=808) | Original Paper (N=606) |
| :--- | :--- | :--- | :--- |
| **$R_M-R_F$** | **Mean** | **1.1236** | **0.50** |
| | **Std dev.** | **6.7130** | **4.49** |
| **SMB** | **Mean** | **0.3675** | **0.29** |
| | **Std dev.** | **4.7166** | **3.07** |
| **HML** | **Mean** | **0.5323** | **0.37** |
| | **Std dev.** | **5.2197** | **2.88** |
| **RMW** | **Mean** | **0.5174** | **0.25** |
| | **Std dev.** | **3.4635** | **2.14** |
| **CMA** | **Mean** | **0.4599** | **0.33** |
| | **Std dev.** | **3.5416** | **2.01** |
| **RF** | **Mean** | **0.6766** | N/A |
| | **Std dev.** | **1.4309** | N/A |

**Analysis of Discrepancies**

The comparison reveals significant differences between the two samples, all of which can be attributed to the different sample periods.

Sample Size: As noted, my sample is 202 months longer. This extended period (post-December 2013) appears to have a substantial impact on the descriptive statistics.

**Means:** The mean for every single factor is higher in my sample than in the original paper. The differences are particularly stark for:

$R_M−R_F$ (Market): 1.12% vs. 0.50%.

$RMW$ (Profitability): 0.52% vs. 0.25%.

$HML$ (Value): 0.53% vs. 0.37%. 

**This suggests that the market, profitability, and value factors have all had very strong average performance in the period from 2014 to 2025.**

**Distributions (Standard Deviation):** The standard deviation (a proxy for volatility) is also dramatically higher across all factors in the samples.

The volatility of the market ($R_M−R_F$) is 6.71 vs. 4.49.

The volatility of $HML$ is 5.22 vs. 2.88.

The volatility of $SMB$ is 4.72 vs. 3.07.

**Potential Reasons:** The post-2013 data included in my sample covers several periods of extreme market volatility that are not in the paper's 1963-2013 sample. This includes the market shocks from the COVID-19 pandemic, the subsequent high-inflation environment, and the aggressive interest rate hike cycle. These events would explain both the higher average returns (driven by strong recoveries) and the much higher standard deviations seen in my sample's descriptive statistics.

The distributions from the samples, which also show large min and max values (e.g., -38.25% to 35.18% for MKTRF), confirm that the data includes periods of extreme market swings not fully captured in the original paper's 50-year sample.

### (d) Variable Construction:

* **Independent Variables (RHS Factors):**
    * **$R_M-R_F$**: The market excess return when compared to risk free asset return?


    * **SMB (Small Minus Big):** The size factor.

        * What it represents: This factor captures the size premium, which is the historical observation that small-company stocks (small market capitalization) tend to have higher average returns than large-company stocks.
     
            * How it's built:
              * Firms are sorted based on their market capitalization (Size).
              * The factor buys (goes long) a diversified portfolio of small stocks.
              * It sells (goes short) a diversified portfolio of big stocks.
                
          * What it means:
              * A positive SMB return means small-cap stocks outperformed large-cap stocks for that period.
              * A portfolio with a positive SMB beta is tilted toward small-cap stocks.


    * **HML (High Minus Low):** The value factor.

        * What it represents: This factor captures the value premium. This is the historical observation that "value" stocks (which appear cheap) tend to outperform "growth" stocks (which appear expensive).
          
         * How it's built:
             * Firms are sorted based on their Book-to-Market (B/M) ratio (the accounting value of the company divided by its stock market value).
             * A High B/M ratio means the stock is "cheap" (a value stock).
             * A Low B/M ratio means the stock is "expensive" (a growth stock).
             * The factor buys (goes long) a portfolio of High B/M (value) stocks.
             * It sells (goes short) a portfolio of Low B/M (growth) stocks.
           
        * What it means:
            * A positive HML return means value stocks outperformed growth stocks for that period.
            * A portfolio with a positive HML beta is tilted toward value stocks. A negative beta means it's tilted toward growth stocks.


    * **RMW (Robust Minus Weak):** The profitability factor.
 
        * What it represents: This factor captures the profitability premium. This is the observation that firms with high (robust) profitability tend to have higher average returns than firms with low (weak) profitability.
          
        * How it's built:
     
            * Firms are sorted based on their Operating Profitability (OP), which the paper measures as revenues minus costs, all divided by book equity.
            * The factor buys (goes long) a portfolio of Robust Profitability stocks.
            * It sells (goes short) a portfolio of Weak Profitability stocks.
              
        * What it means:
            * A positive RMW return means high-profitability firms outperformed low-profitability firms.
            * A portfolio with a positive RMW beta is tilted toward "quality" or high-profitability firms.

     
    * **CMA (Conservative Minus Aggressive):** The investment factor.

        * What it represents: This factor captures the investment premium. This is based on the theory and observation that companies that invest conservatively (grow their assets slowly) tend to have higher returns than firms that invest aggressively (grow their assets very quickly).
 
        * How it's built:
            * Firms are sorted based on their investment rate, which the paper measures as the year-over-year growth in total assets.
            * The factor buys (goes long) a portfolio of low-investment (Conservative) firms.
            * It sells (goes short) a portfolio of high-investment (Aggressive) firms.
              
        * What it means:
            * A positive CMA return means low-investment firms outperformed high-investment firms.
            * A portfolio with a positive CMA beta is tilted toward firms that are more conservative with their asset growth.

### (e) Missing Data:

Missing data is not a concern for this analysis. The descriptive statistics from our samples (N=808) show an identical count for all variables, indicating there are no missing observations (NaN values) within the loaded time-series dataset.

This is consistent with the methodology of the original paper. Fama and French (2015) do not impute missing data; rather, they handle it through sample selection. The samples are constructed by including "all NYSE, AMEX, and NASDAQ stocks on both CRSP and Compustat with...data for Size and B/M". Firms that are missing this required data are simply excluded from the sample before the portfolios are formed.

Because this analysis uses the pre-computed factor returns—which are the result of this rigorous screening process—no further steps were needed to handle missing data.

---

## 3. Replication Analysis

### (a) Model Specification: 

Implement the authors’ main regression specification exactly as
described.

### (b) Estimation Results: 

Present your results in a professional table format comparable
to the original paper’s tables.


### (c) Comparison with Original:

Create side-by-side comparison:

• Your coefficients vs. original paper’s coefficients

• Your standard errors vs. original standard errors

• Your R2 and other fit statistics vs. original

• Statistical significance patterns (which variables are significant in each)


### (d) Replication Assessment: Evaluate success of replication:

• Are coefficient signs the same?

• Are magnitudes similar? Calculate percentage differences

• Are significance levels comparable?

• Overall assessment: successful, partial, or unsuccessful replication


### (e) Discrepancy Investigation: If results differ, investigate potential reasons:

• Different sample periods or data versions

• Missing observations or variables

• Ambiguous specification details in original paper

• Software differences (R vs. Stata vs. Python)

• Errors in original paper or your implementation